### PR TITLE
Docs in spanish

### PR DESCRIPTION
**Pull request by AndreyPootMay**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  |

In this pull-request I updated the docs for spanish documentation.

### CODE REVIEW

File: docs/es/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Yii-Cycle package
+
+Cycle es un motor PHP DataMapper ORM y de modelado de datos diseñado para trabajar con seguridad tanto en aplicaciones PHP clásicas como daemonizadas tales como [RoadRunner](https://github.com/spiral/roadrunner).
```
Comment (terabytesoftw) [inline docs/es/README.md:3]:
```suggestion
Cycle es un motor PHP DataMapper ORM y de modelado de datos diseñado para trabajar tanto en aplicaciones PHP clásicas como daemonizadas tales como [RoadRunner](https://github.com/spiral/roadrunner).
```

File: docs/es/entity-reader.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
+# EntityReader
+
+`EntityReader` es una herramienta útil para transferir de forma segura las solicitudes de selección del repositorio al tiempo de ejecución del usuario.
```
Comment (terabytesoftw) [inline docs/es/entity-reader.md:3]:
```suggestion
`EntityReader` es una herramienta útil para transferir de forma segura las solicitudes de selección del repositorio en tiempo de ejecución del usuario.
```

File: docs/es/entity-reader.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+Una solicitud de selección se refiere a una instancia de uno de las clases: ``CycleORM\Select`` o ``Spiral\Database\Query\SelectQuery``.
```
Comment (terabytesoftw) [inline docs/es/entity-reader.md:5]:
```suggestion
Una solicitud de selección se refiere a una instancia de una de las clases: ``CycleORM\Select`` o ``Spiral\Database\Query\SelectQuery``.
```

File: docs/es/entity-reader.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+Lo que hay que saber sobre `EntityReader`:
+
+* La clase `EntityReader` implementa la interfaz `IteratorAggregate`. Esto permite utilizar el objeto `EntityReader` en un bucle `foreach`.
+* Con `EntityReader` se puede ajustar la consulta de selección pasada:
```
Comment (terabytesoftw) [inline docs/es/entity-reader.md:10]:
```suggestion
* Con `EntityReader` se puede ajustar la consulta de selección:
```

File: docs/es/entity-reader.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+  - Establezca `Limit` y `Offset` manualmente o con `OffsetPaginator`.
```
Comment (terabytesoftw) [inline docs/es/entity-reader.md:11]:
```suggestion
  - Establezca `Limit` y `Offset` manualmente con `OffsetPaginator`.
```

File: docs/es/entity-reader.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+  - Establece la clasificación. Pero tenga en cuenta que la ordenación por `EntityReader` no sustituye la ordenación en la consulta original, sólo la complementa.Sin embargo, cada llamada posterior al método `withSort()` sustituirá la configuración de clasificación del objeto `EntityReader`.
```
Comment (terabytesoftw) [inline docs/es/entity-reader.md:12]:
```suggestion
  - La ordenación por `EntityReader` no sustituye la ordenación en la consulta original, sólo la complementa. Sin embargo, cada llamada posterior al método `withSort()` sustituirá la configuración de clasificación del objeto `EntityReader`.
```

File: docs/es/entity-reader.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+  - Aplicar filtro. Las condiciones del filtro `EntityReader` tampoco sustituyen al de filtrado en la consulta original, la complementan. Así, al filtrar en el objeto `EntityReader`, sólo puede refinar la selección pero no ampliarla.
```
Comment (terabytesoftw) [inline docs/es/entity-reader.md:13]:
```suggestion
  - Las condiciones del filtro `EntityReader` tampoco sustituyen al de filtrado en la consulta original, solo la complementan. Así, que al filtrar el objeto `EntityReader`, sólo puede ajustar la selección pero no ampliarla.
```

File: docs/es/entity-reader.md [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+* `EntityReader` no extrae los datos de la base de datos de una sola vez. Sólo accede a la base de datos cuando se consultan esos datos.
+* Si utiliza los métodos `read()` y `readOne()` para leer los datos, entonces el `EntityReader` lo almacenará en una caché. El resultado de una llamada a `count()` también se almacena en caché.
```
Comment (terabytesoftw) [inline docs/es/entity-reader.md:15]:
```suggestion
* Si utiliza los métodos `read()` y `readOne()` para leer los datos, entonces `EntityReader` lo almacenará en una caché. El resultado de una llamada a `count()` también se almacena en caché.
```

File: docs/es/entity-reader.md [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+* El método `count()` devuelve el número de todos los elementos de la muestra sin tener en cuenta las restricciones `Limit` y `Offset`.
+* Si no quieres que los datos se almacenen en caché, utiliza el método `getIterator()`.Sin embargo, si la caché ya está llena, `getIterator()` devolverá el contenido de la caché.
```
Comment (terabytesoftw) [inline docs/es/entity-reader.md:17]:
```suggestion
* Si no quieres que los datos se almacenen en caché, utiliza el método `getIterator()`. Sin embargo, si la caché ya está llena, `getIterator()` devolverá el contenido de la caché.
```

File: docs/es/entity-reader.md [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+
+## Ejemplos
+
+Vamos a implementar un repositorio para trabajar con la tabla de artículos. Queremos un método para obtener artículos públicos `findPublic()` pero
```
Comment (terabytesoftw) [inline docs/es/entity-reader.md:21]:
```suggestion
Vamos a implementar un repositorio para trabajar con la tabla de artículos. Queremos un método para obtener artículos públicos `findPublic()`
```

File: docs/es/entity-reader.md [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+    count($lastPublicReader->read()),
+    $lastPublicReader->count()
+);
+foreach ($lastPublicReader->read() as $article) {
+    // ...
+}
+```
+
+La ordenación a través de `EntityReader` no sustituye la ordenación en la consulta inicial, sino que le añade algo más.
+Si necesitas establecer la ordenación por defecto en un método del repositorio pero quieres poder cambiarla en un controlador, puedes puede hacerlo de la siguiente manera:
```
Comment (terabytesoftw) [inline docs/es/entity-reader.md:116]:
```suggestion
Si necesitas establecer la ordenación por defecto en un método del repositorio, pero quieres poder cambiarla en un controlador, puedes puede hacerlo de la siguiente manera:
```

File: docs/es/entity-reader.md [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+function index(ArticleRepository $repository)
+{
+    $articlesReader = $repository
+        // Obteniendo EntityReader
+        ->findPublic()
+        // Aplicando nueva ordenación.
+        ->withSort(Sort::any()->withOrder(['published_at' => 'asc']));
+}
+```
+Puede refinar las condiciones de consulta con filtros. Estas condiciones de filtrado se añaden a las condiciones de consulta de selección originales, pero NO las sustituyen.
```
Comment (terabytesoftw) [inline docs/es/entity-reader.md:146]:
```suggestion
Puede ajustar las condiciones de consulta con filtros. Estas condiciones de filtrado se añaden a las condiciones de consulta de selección originales, pero no las sustituyen.
```

File: docs/es/installation.md [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+
+La forma preferida de instalar este paquete es a través de [Composer](https://getcomposer.org/download/):
+
+```bash
+composer require yiisoft/yii-cycle
+```
+
+## Configurar paquete
+
+Si utiliza Yii con `composer-config-plugin`, la configuración de Yii-Cycle se puede especificar en `config/params.php`:
```
Comment (terabytesoftw) [inline docs/es/installation.md:11]:
```suggestion
Si utiliza `yiisoft/config`, la configuración de `yisoft/yii-cycle` se debe especificar en `config/params.php`:
```

File: docs/es/reading-schema.md [hunk 14]
```diff
@@ -0,0 +1,137 @@
+# Reading DB schema
+
+Cycle ORM se basa en el esquema de la base de datos - objeto, que implementa la interfaz `\Cycle\ORM\SchemaInterface`.
```
Comment (terabytesoftw) [inline docs/es/reading-schema.md:3]:
```suggestion
Cycle ORM se basa en un esquema de base de datos, que implementa la interfaz `\Cycle\ORM\SchemaInterface`.
```

File: docs/es/reading-schema.md [hunk 15]
```diff
@@ -0,0 +1,137 @@
...
+
+Dado que un esquema se construye a partir de un array de una determinada estructura, podemos almacenarlo en caché o en un archivo de texto.
```
Comment (terabytesoftw) [inline docs/es/reading-schema.md:5]:
```suggestion
Dado que el esquema se construye a partir de un arreglo con una estructura definida, podemos almacenarlo en caché o en un archivo de texto.
```

File: docs/es/reading-schema.md [hunk 16]
```diff
@@ -0,0 +1,137 @@
...
+
+Puede mostrar el esquema utilizado actualmente ejecutando el comando `cycle/schema`.
+
+En el paquete `yii-cycle` el esquema puede ser construido desde múltiples fuentes representadas por múltiples proveedores que implementan
```
Comment (terabytesoftw) [inline docs/es/reading-schema.md:9]:
```suggestion
En el paquete `yiisoft/yii-cycle` el esquema puede ser construido desde múltiples fuentes representadas por múltiples proveedores que implementan
```

File: docs/es/reading-schema.md [hunk 17]
```diff
@@ -0,0 +1,137 @@
...
+`SchemaProviderInterface`.
+
+Para utilizar varios proveedores de esquemas a su vez, se utiliza el proveedor `SchemaProviderPipeline` de agrupación.
```
Comment (terabytesoftw) [inline docs/es/reading-schema.md:12]:
```suggestion
Para utilizar varios proveedores de esquemas, se utiliza el proveedor `SchemaProviderPipeline` de agrupación.
```

File: docs/es/reading-schema.md [hunk 18]
```diff
@@ -0,0 +1,137 @@
...
+Puede configurar este proveedor en la sección `schema-providers` de un archivo `config/params.php`.
```
Comment (terabytesoftw) [inline docs/es/reading-schema.md:13]:
```suggestion
Puede configurar este proveedor en la sección `schema-providers` en un archivo `config/params.php`.
```

File: docs/es/reading-schema.md [hunk 19]
```diff
@@ -0,0 +1,137 @@
...
+Organice los proveedores de esquemas en un orden tal, que los proveedores de caché estén al principio de la lista y los proveedores de esquemas de origen al final.
```
Comment (terabytesoftw) [inline docs/es/reading-schema.md:14]:
```suggestion
Los proveedores de esquemas deben estar organizados de la siguiente manera, los proveedores de caché deben estar al principio de la lista y los proveedores de esquemas de origen al final.
```

File: docs/es/reading-schema.md [hunk 20]
```diff
@@ -0,0 +1,137 @@
...
+
+
+## Esquema basado en la anotación de entidades
+
+Por defecto, el esquema se construye en base a las anotaciones que se encuentran en las entidades de su proyecto.
+
+Cuando se construye un esquema, los generadores se ejecutan secuencialmente. La secuencia se determina en una instancia de
+`SchemaConveyorInterface`. Puede insertar sus propios generadores en este transportador definiéndolos en
```
Comment (terabytesoftw) [inline docs/es/reading-schema.md:22]:
```suggestion
`SchemaConveyorInterface`. Puede insertar sus propios generadores dentro del transpotador, para ello debe especificarlos en
```

File: docs/es/reading-schema.md [hunk 21]
```diff
@@ -0,0 +1,137 @@
...
+`entity-paths` una opción del archivo `config/params.php`.
```
Comment (terabytesoftw) [inline docs/es/reading-schema.md:23]:
```suggestion
`entity-paths` dentro del archivo `config/params.php`.
```

File: docs/es/reading-schema.md [hunk 22]
```diff
@@ -0,0 +1,137 @@
...
+Para obtener un esquema del transportador se usa la clase `FromConveyorSchemaProvider`.
+
+El proceso de construcción de esquemas a partir de anotaciones es relativamente pesado en términos de rendimiento. Por lo tanto, en caso de
+usar anotaciones es una buena idea usar el caché de esquemas.
+
+## Esquemas desde caché
+
+La lectura y escritura de un esquema desde y hacia la caché ocurre en `SimpleCacheSchemaProvider`.
+
+Colócalo al principio de la lista de proveedores para que el proceso de obtención de un esquema sea significativamente más rápido.
```
Comment (terabytesoftw) [inline docs/es/reading-schema.md:34]:
```suggestion
Debe indicarse al principio de la lista de proveedores para que el proceso de obtención de un esquema sea significativamente más rápido.
```

File: docs/es/reading-schema.md [hunk 23]
```diff
@@ -0,0 +1,137 @@
...
+           'id' => 'int'
+        ],
+        Schema::RELATIONS   => []
+    ]
+];
+```
+
+Tenga en cuenta que:
+
+1. `FromFilesSchemaProvider` carga un esquema desde archivos PHP a través de `include`. Esto requiere precauciones de seguridad.
```
Comment (terabytesoftw) [inline docs/es/reading-schema.md:78]:
```suggestion
1. `FromFilesSchemaProvider` carga un esquema desde un archivo `PHP` a través de `include`. Esto requiere precauciones de seguridad.
```

File: docs/es/reading-schema.md [hunk 24]
```diff
@@ -0,0 +1,137 @@
...
+   Asegúrese de almacenar el archivo del esquema en una ruta segura y restringida a los usuarios.
+2. Puede especificar varios archivos de esquema, que se fusionarán en un solo esquema.
```
Comment (terabytesoftw) [inline docs/es/reading-schema.md:80]:
```suggestion
2. Puede especificar varios archivos de esquema, los cuales se fusionarán en un solo esquema.
```

File: docs/es/reading-schema.md [hunk 25]
```diff
@@ -0,0 +1,137 @@
...
+Se lanzará una excepción en caso de colisión de roles.
+
+3. Gracias a la caché interna, la carga del esquema desde un archivo PHP es tan rápida que puede omitirse por completo la caché externa.
```
Comment (terabytesoftw) [inline docs/es/reading-schema.md:83]:
```suggestion
3. Gracias a la caché interna, la carga del esquema desde un archivo `PHP` es tan rápida que puede omitirse por completo la caché externa.
```

File: docs/es/reading-schema.md [hunk 26]
```diff
@@ -0,0 +1,137 @@
...
+Pero en el caso de que se carguen varios archivos, puede tardar más tiempo en fusionarlos.
```
Comment (terabytesoftw) [inline docs/es/reading-schema.md:84]:
```suggestion
En el caso de que se carguen varios archivos, puede tardar más tiempo en fusionase.
```

File: docs/es/reading-schema.md [hunk 27]
```diff
@@ -0,0 +1,137 @@
...
+4. No se pueden generar migraciones basadas en el esquema de archivos PHP. [Para más información mire el issue #25](https://github.com/yiisoft/yii-cycle/issues/25)
```
Comment (terabytesoftw) [inline docs/es/reading-schema.md:85]:
```suggestion
4. No se pueden generar migraciones basadas en el esquema de archivos `PHP`. [Para más información mire el issue #25](https://github.com/yiisoft/yii-cycle/issues/25)
```

File: docs/es/reading-schema.md [hunk 28]
```diff
@@ -0,0 +1,137 @@
...
+        ],
+    ]
+];
+```
+
+## Pasar de las anotaciones al archivo
+
+### Comando de consola
+
+Para exportar el esquema como archivo PHP se puede utilizar el comando `cycle/schema/php`.
```
Comment (terabytesoftw) [inline docs/es/reading-schema.md:117]:
```suggestion
Para exportar el esquema como archivo `PHP` se puede utilizar el comando `cycle/schema/php`.
```

File: docs/es/reading-schema.md [hunk 29]
```diff
@@ -0,0 +1,137 @@
...
+Especifique un nombre de archivo como argumento y el esquema se escribirá en él:
```
Comment (terabytesoftw) [inline docs/es/reading-schema.md:118]:
```suggestion
Especifique el nombre del archivo como argumento y el esquema se escribirá en él:
```

File: docs/es/reading-schema.md [hunk 30]
```diff
@@ -0,0 +1,137 @@
...
+
+```bash
+cycle/schema/php @runtime/schema.php
+```
+
+`@runtime` se sustituye automáticamente. El esquema se exportará al archivo `schema.php`.
+
+Asegúrese de que el esquema exportado es correcto y luego pase a utilizarlo mediante `FromFilesSchemaProvider`.
```
Comment (terabytesoftw) [inline docs/es/reading-schema.md:126]:
```suggestion
Asegúrese de que el esquema exportado es correcto y luego puede utilizarlo mediante `FromFilesSchemaProvider`.
```

File: docs/es/reading-schema.md [hunk 31]
```diff
@@ -0,0 +1,137 @@
...
+
+Se pueden combinar ambas formas para describir un esquema. Durante el desarrollo de un proyecto es muy útil utilizar anotaciones. Puede generar
+migraciones basadas en ellas. Para el uso en producción, el esquema puede ser trasladado a un archivo.
+
+### Provider `PhpFileSchemaProvider`
+
+A diferencia de `FromFilesSchemaProvider`, el `PhpFileSchemaProvider` trabaja con un solo archivo. pero, `PhpFileSchemaProvider`
+no sólo puede leer el esquema, sino también guardarlo.
+
+En el modo de lectura y escritura de un archivo de esquema, el proveedor `PhpFileSchemaProvider` funciona de forma similar a la caché, con
```
Comment (terabytesoftw) [inline docs/es/reading-schema.md:136]:
```suggestion
En el modo de lectura y escritura un archivo de esquema que utilice el proveedor `PhpFileSchemaProvider`, funcionara de forma similar a la caché, con
```

File: docs/es/reading-schema.md [hunk 32]
```diff
@@ -0,0 +1,137 @@
...
+la única diferencia es que el resultado guardado (archivo de esquema), puede ser guardado en codebase.
```
Comment (terabytesoftw) [inline docs/es/reading-schema.md:137]:
```suggestion
la única diferencia que el resultado guardado (archivo de esquema), puede ser guardado en codebase.
```

**Review by terabytesoftw — CHANGES_REQUESTED** [review]

File: docs/es/reading-schema.md [hunk 33]
```diff
@@ -0,0 +1,137 @@
...
+`@runtime` se sustituye automáticamente. El esquema se exportará al archivo `schema.php`.
+
+Asegúrese de que el esquema exportado es correcto y luego pase a utilizarlo mediante `FromFilesSchemaProvider`.
+
+Se pueden combinar ambas formas para describir un esquema. Durante el desarrollo de un proyecto es muy útil utilizar anotaciones. Puede generar
+migraciones basadas en ellas. Para el uso en producción, el esquema puede ser trasladado a un archivo.
+
+### Provider `PhpFileSchemaProvider`
+
+A diferencia de `FromFilesSchemaProvider`, el `PhpFileSchemaProvider` trabaja con un solo archivo. pero, `PhpFileSchemaProvider`
```
Comment (terabytesoftw) [inline docs/es/reading-schema.md:133]:
```suggestion
A diferencia de `FromFilesSchemaProvider`, el archivo `PhpFileSchemaProvider` trabaja con un solo archivo de esquema `PhpFileSchemaProvider`
```

**Review by terabytesoftw — CHANGES_REQUESTED** [review]

**Comment by AndreyPootMay** [comment]
Ok, I'm going to change the files correctly, thanks :3

**Comment by terabytesoftw** [comment]
👍

**Pull request merged**